Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -528,16 +539,16 @@ class CombinatorialKalmanFilter {
         const auto& boundParams = std::get<BoundTrackParameters>(boundState);
 
         // Get all source links on the surface
-        const auto& sourcelinks = sourcelink_it->second;
-
+        auto [lower_it, upper_it] =
+            m_sourcelinkAccessor.range(surface->geometryId());
         // Calibrate all the source links on the surface since the selection has
         // to be done based on calibrated measurement
-        std::vector<BoundVariantMeasurement<source_link_t>> measurements;
-        measurements.reserve(sourcelinks.size());
-        std::transform(sourcelinks.begin(), sourcelinks.end(),
-                       std::back_inserter(measurements), [&](const auto& sl) {
-                         return m_calibrator(sl, boundParams);
-                       });
+        std::vector<BoundVariantMeasurement<SourceLink>> measurements;
```

Comment:
Eventually I suppose this vector should go into the state or something like that, so we can avoid having to reallocate this every time.